src/Symfony/Component/Translation/Tests/TranslatorTest.php

```diff
@@ -239,6 +239,30 @@ public function testWhenAResourceHasNoRegisteredLoader()
         $translator->trans('foo');
     }
 
+    public function testFallbackCatalogueResources()
+    {
+        $translator = new Translator('en_GB', new MessageSelector());
+        $translator->addLoader('yml', new \Symfony\Component\Translation\Loader\YamlFileLoader());
+        $translator->addResource('yml', __DIR__.'/fixtures/empty.yml', 'en_GB');
+        $translator->addResource('yml', __DIR__.'/fixtures/resources.yml', 'en');
+
+        // force catalogue loading
+        $this->assertEquals('bar', $translator->trans('foo', array()));
+
+        $cataloguesProperty = new \ReflectionProperty($translator, 'catalogues');
```

Comment:
you can get catalogue from the translator without using reflection, [$translator->getCatalogue('en')](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Translation/Translator.php#L223)